client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -377,6 +378,12 @@ function mediaButton( editor ) {
 				editor.fire( 'SetTextAreaContent', { content } );
 			}
 
+			// We have just replaced some image nodes, potentially losing selection/focus.
+			// So if an image node was selected and one isn't now, re-select it.
```

Comment:
Yes.. there is other code that replaces or appends depending on whether an image is selected. That code is intended to match usability behavior of core.